os/kernel/log_dump/log_dump.c

```diff
@@ -155,7 +155,11 @@ int log_dump_save(char ch)
 
 	size_t free_size;
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
-	free_size = kmm_get_heap_free_size(void);
+#ifdef CONFIG_BUILD_PROTECTED
+	free_size = kmm_get_heap_free_size();
```

Comment:
`kmm_get_heap_free_size()` will be the same as `umm_get_heap_free_size()` for flat build.
So I think we don't need to separate this.

For `malloc` case, we can use `kmm_malloc()` from flat build, but `kmm_malloc()` will work properly.

Reply:
agree